### ISSUE TITLE
Error in documentation for ESP8266WiFiGenericClass::hostByName

### ISSUE THREAD

**Issue by UlliBien** [esp8266/Arduino#3298]
#### Hardware
Core Version:      	2.3.0

### Description
Ducomentation says:

>  * @return 1 if aIPAddrString was successfully converted to an IP address,
>  *          else error code

In case of an error 0 is returned. No error code

**Comment by devyte** [comment]
Where in the docs? Or do you mean the comments in the source?

**Comment by UlliBien** [comment]
Sorry for my bad explanation.
It's in ESP8266WiFiGeneric.cpp, line 410 ff.

```
/**
 * Resolve the given hostname to an IP address.
 * @param aHostname     Name to be resolved
 * @param aResult       IPAddress structure to store the returned IP address
 * @return 1 if aIPAddrString was successfully converted to an IP address,
 *          else error code
 */
int ESP8266WiFiGenericClass::hostByName(const char* aHostname, IPAddress& aResult) {
...
```

The code of the method ends with
`    return (err == ERR_OK) ? 1 : 0;`

which returns 1 or 0, but not an error code.

**Comment by earlephilhower** [comment]
Fixed via #5325
